### PR TITLE
Fix #32799: Use correct components for ad-blocking

### PR DESCRIPTION
**Pull request by cuba**
Resolves https://github.com/brave/brave-browser/issues/32799

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_shields/browser/ad_block_component_installer.cc [hunk 1]
```diff
@@ -48,17 +48,16 @@ const char kAdBlockFilterListCatalogComponentBase64PublicKey[] =
...
+const char kAdBlockDefaultComponentName[] = "Brave Ad Block Updater";
+const char kAdBlockDefaultComponentId[] = "iodkpdagapdfkphljnddpjlldadblomo";
+const char kAdBlockDefaultComponentBase64PublicKey[] =
+    "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAsD/B/MGdz0gh7WkcFARn"
+    "ZTBX9KAw2fuGeogijoI+fET38IK0L+P/trCT2NshqhRNmrDpLzV2+Dmes6PvkA+O"
+    "dQkUV6VbChJG+baTfr3Oo5PdE0WxmP9Xh8XD7p85DQrk0jJilKuElxpK7Yq0JhcT"
+    "Sc3XNHeTwBVqCnHwWZZ+XysYQfjuDQ0MgQpS/s7U04OZ63NIPe/iCQm32stvS/pE"
+    "ya7KdBZXgRBQ59U6M1n1Ikkp3vfECShbBld6VrrmNrl59yKWlEPepJ9oqUc2Wf2M"
+    "q+SDNXROG554RnU4BnDJaNETTkDTZ0Pn+rmLmp1qY5Si0yGsfHkrv3FS3vdxVozO"
+    "PQIDAQAB";
```
Comment (antonok-edm) [inline components/brave_shields/browser/ad_block_component_installer.cc:60]: We already have these [in ad_block_service.cc](https://github.com/brave/brave-core/blob/6f7a6f8283db4c405f095df6fdf07405c8de638a/components/brave_shields/browser/ad_block_service.cc#L38C1-L47) - can you deduplicate them? I don't have a preference as to where; here should be fine, as long as it's just one.

**Review by antonok-edm — COMMENTED** [review]
Looks alright to me other than one request

File: components/brave_shields/browser/ad_block_component_installer.cc (hunk 1 above)
Comment (cuba) [reply components/brave_shields/browser/ad_block_component_installer.cc]: Ok moved it to a shared file

File: components/brave_shields/browser/ad_block_service.cc [hunk 2]
```diff
@@ -16,6 +16,7 @@
 #include "base/threading/thread_restrictions.h"
 #include "brave/components/brave_shields/adblock/rs/src/lib.rs.h"
 #include "brave/components/brave_shields/browser/ad_block_component_filters_provider.h"
+#include "brave/components/brave_shields/browser/ad_block_component_installer.h"
```
Comment (antonok-edm) [inline components/brave_shields/browser/ad_block_service.cc:19]: needed? it's the only change in the file

File: ios/browser/api/brave_shields/adblock_filter_list_catalog_entry.h [hunk 3]
```diff
@@ -23,9 +23,6 @@ OBJC_EXPORT
 @property(readonly) bool firstPartyProtections;
 @property(readonly) uint8_t permissionMask;
 @property(readonly) NSString* componentId;
-@property(readonly) NSString* base64PublicKey;
```
Comment (antonok-edm) [inline ios/browser/api/brave_shields/adblock_filter_list_catalog_entry.h:26]: we might still want this one?

**Review by antonok-edm — APPROVED** [review]
awesome!

File: ios/browser/api/brave_shields/adblock_filter_list_catalog_entry.h (hunk 3 above)
Comment (cuba) [reply ios/browser/api/brave_shields/adblock_filter_list_catalog_entry.h:26]: It's just the iOS interface. I don't see why we need this exposed to iOS. But maybe I'll need it in the future. I don't know. I'll put it back

File: components/brave_shields/browser/ad_block_component_installer.h [hunk 4]
```diff
@@ -11,6 +11,17 @@
 #include "base/files/file_path.h"
 #include "base/functional/callback.h"
 
+const char kAdBlockDefaultComponentName[] = "Brave Ad Block Updater";
```
[the remark this answers is not in the document]
Comment (cuba) [reply components/brave_shields/browser/ad_block_component_installer.h:14]: Actually can remove this from the header altogether as desktop is no longer using this. We'll eventually remove this too when switching to the unified list

**Review by Brandon-T — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.61.45 Chromium: 118.0.5993.96 (Official Build) nightly (64-bit)
-- | --
Revision | 31c36588a37ea88c2276d901ebd20dbf2344fde3
OS | Windows 11 Version 22H2 (Build 22621.2428)
```

Using the STR/information outlined via https://github.com/brave/brave-browser/issues/32799#issue-1884399665, ensured that `ad-blocking` was generally working as per the following:

**`Quick Note:`** If this regressed general blocking, we would have heard it by now from Nighty users within Brave.

* visited https://www.zerohedge.com and disabled shields and ensured that `ads` loaded within the page
  * re-enabled shields and ensured that ads were being blocked without issues
* visited https://www.google.ca and searched for Under Armour and ensured that first-party ads were appearing
  * switched to `Aggressively block tracker & ads` and ensured those ads were being removed/not being displayed anymore
* visited https://www.blizzard.com/en-gb/ and ensured that enabling `Block scripts` blocked the carousel
* visited https://digikey.com and ensured there was no `cookie consent` banner being displayed at the top of the page
  * https://user-images.githubusercontent.com/2602313/245333866-0b9ce862-7c52-4381-af8b-a593ee32c3a3.png
* visited https://mixed-script.badssl.com and ensured the page appears `grey`

**Pull request merged**